httpcore5-h2/src/main/java/org/apache/hc/core5/http2/impl/nio/ClientHttp2StreamMultiplexer.java

```diff
@@ -83,10 +83,15 @@ public ClientHttp2StreamMultiplexer(
     Http2StreamHandler createRemotelyInitiatedStream(
             final Http2StreamChannel channel,
             final HttpProcessor httpProcessor,
-            final BasicHttpConnectionMetrics connMetrics) throws IOException {
+            final BasicHttpConnectionMetrics connMetrics,
+            final Http2StreamHandler originalHandler) throws IOException {
         final HttpCoreContext context = HttpCoreContext.create();
         context.setAttribute(HttpCoreContext.SSL_SESSION, getSSLSession());
         context.setAttribute(HttpCoreContext.CONNECTION_ENDPOINT, getEndpointDetails());
+        if (originalHandler != null && originalHandler instanceof ClientHttp2StreamHandler) {
```

Comment:
@ato I agree in principle but can this cast be avoided?

Reply:
Oops. I'll add a `getContext()` method to Http2StreamHandler. I mistakenly thought it was a public interface and was trying to avoid a breaking change. I should have read the code more carefully, it's package private.